packages/pieces/community/pinterest/src/lib/common/props.ts

```diff
@@ -0,0 +1,160 @@
+import { DropdownOption, Property } from '@activepieces/pieces-framework';
+import { makeRequest } from '.';
+import { HttpMethod } from '@activepieces/pieces-common';
+
+export const boardIdDropdown = Property.Dropdown({
+  displayName: 'board Id',
+  required: true,
+  refreshers: ['ad_account_id'],
+  options: async ({ auth, ad_account_id }) => {
+    if (!auth) {
+      return {
+        disabled: true,
+        placeholder: 'Please connect your account',
+        options: [],
+      };
+    }
+    let path = '/boards';
+    if (ad_account_id) {
+      path = `/boards?ad_account_id=${ad_account_id}`;
+    }
+    const apiKey = auth as string;
+    const boards = await makeRequest(apiKey, HttpMethod.GET, path);
+
+    const options: DropdownOption<string>[] = boards.items.map((board: any) => ({
+      label: board.name,
+      value: board.id,
+    }));
+
+    return {
+      disabled: false,
+      options,
+    };
+  },
+});
+
+export const pinIdDropdown = Property.Dropdown({
+  displayName: 'pin Id',
+  required: true,
+  refreshers: ['auth'],
+  options: async ({ auth }) => {
```

Comment:
Error Handling: Add proper error handling in dropdown options:


```suggestion
     options: async ({ auth, ad_account_id }) => {
     if (!auth) {
       return {
         disabled: true,
         placeholder: 'Please connect your account',
         options: [],
       };
     }
     try {
       // ... existing code
     } catch (error) {
       return {
         disabled: true,
         placeholder: 'Error loading boards. Please try again.',
         options: [],
       };
     }
   }
```